[CI] MixedClusterClientYamlTestSuiteIT test {p0=search.vectors/42_knn_search_int4_flat/Vector similarity with filter only} failing

**Build Scans:**
- [elasticsearch-periodic #6153 / 8.19.0_bwc](https://gradle-enterprise.elastic.co/s/ge6uv7cb5hdw4)
- [elasticsearch-pull-request #54250 / 9.0.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/564kc7nqsb4mu)
- [elasticsearch-pull-request #54231 / 9.0.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/tvxbysa4s2eva)
- [elasticsearch-pull-request #54218 / 8.19.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/3pl4r6e3msoz6)
- [elasticsearch-pull-request #54183 / 9.0.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/xzqhpebmjzmiy)
- [elasticsearch-pull-request #54176 / 8.19.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mib76rdt3f3om)
- [elasticsearch-pull-request #54167 / 9.0.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mm7aaezmskqfa)
- [elasticsearch-intake #16890 / 9.0.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mzuescbjz4nxe)
- [elasticsearch-intake #16889 / 9.0.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ojwdgnjdyzjow)
- [elasticsearch-pull-request #54100 / 8.18.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/gwpuiiy4xrro4)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.19.0#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/42_knn_search_int4_flat/Vector similarity with filter only}" -Dtests.seed=11B68209F9513935 -Dtests.bwc=true -Dtests.locale=ii-Yiii-CN -Dtests.timezone=Atlantic/South_Georgia -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dsearch.vectors%2F42_knn_search_int4_flat%2FVector%20similarity%20with%20filter%20only%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [search.vectors/42_knn_search_int4_flat:228]: field [hits.hits] doesn't have length [1]
Expected: <1>
     but: was <0>
```

**Issue Reasons:**
- [main] 10 failures in test test {p0=search.vectors/42_knn_search_int4_flat/Vector similarity with filter only} (2.2% fail rate in 456 executions)
- [main] 2 failures in step 8.19.0_bwc-snapshots (1.0% fail rate in 193 executions)
- [main] 6 failures in step 9.0.0_bwc-snapshots (3.5% fail rate in 173 executions)
- [main] 7 failures in pipeline elasticsearch-pull-request (4.5% fail rate in 157 executions)
- [main] 2 failures in pipeline elasticsearch-intake (2.8% fail rate in 72 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -329,9 +329,6 @@ tests:
 - class: org.elasticsearch.ingest.geoip.FullClusterRestartIT
   method: testGeoIpSystemFeaturesMigration {cluster=UPGRADED}
   issue: https://github.com/elastic/elasticsearch/issues/121115
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search.vectors/42_knn_search_int4_flat/Vector similarity with filter only}
-  issue: https://github.com/elastic/elasticsearch/issues/121412
 - class: org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT
   method: test {yaml=reference/cat/health/cat-health-no-timestamp-example}
   issue: https://github.com/elastic/elasticsearch/issues/121867

